lib/query/builder.js

```diff
@@ -1008,35 +1008,44 @@ assign(Builder.prototype, {
     return this;
   },
 
-  // Remove everything from select clause
+  // Deprecated. Remove everything from select clause
   clearSelect() {
     this._clearGrouping('columns');
     return this;
   },
 
-  // Remove everything from where clause
+  // Deprecated. Remove everything from where clause
   clearWhere() {
     this._clearGrouping('where');
     return this;
   },
 
-  // Remove everything from group clause
+  // Deprecated. Remove everything from group clause
   clearGroup() {
     this._clearGrouping('group');
     return this;
   },
 
-  // Remove everything from order clause
+  // Deprecated. Remove everything from order clause
   clearOrder() {
     this._clearGrouping('order');
     return this;
   },
 
-  // Remove everything from having clause
+  // Deprecated. Remove everything from having clause
```

Comment:
Why mark existing methods as deprecated?

Reply:
For further refactoring, remove duplicate features.
Leave backward compatibility to major version.
What do you think?

Reply:
Isn't there a difference between those methods, though? From my understanding, existing ones are more granular, while new ones clears everything. Do I read code wrong?

Reply:
`clear("order")` method completely replaces `clearOrder()`, the difference is that clear gives you more options for limit, offset and other